bundles/jcr/repoinit/src/main/java/org/apache/sling/jcr/repoinit/impl/AclUtil.java

```diff
@@ -122,21 +197,60 @@ private static String toString(JackrabbitAccessControlEntry entry) throws Reposi
         private final Principal principal;
         private final Privilege[] privileges;
         private final boolean isAllow;
-        
+        private final LocalRestrictions oakRestrictions;
+
         LocalAccessControlEntry(Principal principal, Privilege[] privileges, boolean isAllow) {
+            this(principal, privileges, isAllow, null);
+        }
+
+        LocalAccessControlEntry(Principal principal, Privilege[] privileges,
+                                boolean isAllow, LocalRestrictions oakRestrictions) {
             this.principal = principal;
             this.privileges = privileges;
             this.isAllow = isAllow;
+            this.oakRestrictions = oakRestrictions != null ? oakRestrictions : new LocalRestrictions();
         }
         
         public boolean isContainedIn(JackrabbitAccessControlEntry other) throws RepositoryException {
             return other.getPrincipal().equals(principal) &&
                     contains(other.getPrivileges(), privileges) &&
                     other.isAllow() == isAllow &&
-                    ( other.getRestrictionNames() == null || 
-                        other.getRestrictionNames().length == 0 );
+                    sameRestrictions(other);
         }
-        
+
+        /**
+         * compares if restrictions present in jackrabbit access control entry
+         * is same as specified restrictions in repo init
+         * @param jace
+         * @return
+         * @throws RepositoryException
+         */
+        private boolean sameRestrictions(JackrabbitAccessControlEntry jace) throws RepositoryException {
+            // total (multivalue and simple)  number of restrictions should be same
+            if(jace.getRestrictionNames().length == (oakRestrictions.size())){
+                for(String rn : jace.getRestrictionNames()){
+                    Value[] oldValues = jace.getRestrictions(rn);
+                    Value[] newValues = oakRestrictions.getRestrictions().get(rn) != null
+                                        ? new Value[]{oakRestrictions.getRestrictions().get(rn)}
+                                        : oakRestrictions.getMVRestrictions().get(rn);
+                    if((newValues == null || newValues.length == 0) && (oldValues == null || oldValues.length == 0)){
+                        continue;
+                    }
+
+                    if(newValues.length != oldValues.length){
+                        return false;
+                    }
+                    for(int i=0;i<newValues.length;i++){
+                        if(!newValues[i].equals(oldValues[i])){
+                            return false;
+                        }
+                    }
+                }
+                return true;
+            }
+            return false;
+        }
+
         private boolean contains(Privilege[] first, Privilege[] second) {
             // we need to ensure that the privilege order is not taken into account, so we use sets
```

Comment:
👍  I will log a bug for this since it is unrelated to this change. Thanks for the test case